pyomo/repn/plugins/gams_writer.py

```diff
@@ -326,15 +333,17 @@ def _write_model(self,
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
-                    if v.bounds == (0,1):
+                    if (v.has_lb() and (value(v.lb) >= 0)) and \
+                       (v.has_ub() and (value(v.ub) <= 1)):
                         binary.append(var)
                     else:
                         ints.append(var)
-                elif v.lb == 0:
+                elif value(v.lb) == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
             else:
+                # GH: doesn't seem like this block is ever hit
```

Comment:
I've verified that there are solver tests that use Expression (and related objects) in their constraints (e.g., [here](https://github.com/gseastream/pyomo/blob/gams-writer/pyomo/solvers/tests/models/LP_simple.py#L88)). Mind posting a simple example that reproduces this?